lib/bunyan.js

```diff
@@ -522,6 +532,9 @@ Logger.prototype.addStream = function addStream(s, defaultLevel) {
     if (!s.type) {
         if (s.stream) {
             s.type = 'stream';
+            if (!isWritable(s.stream)) {
+                throw TypeError('type "stream" stream must be writable');
```

Comment:
Let's use `assert.ok` (as is done lower in this function) instead of the `if`-block. Untested:

```
assert.ok(isWritable(s.stream), '"stream" stream is not writable: ' + util.inspect(s.stream));
```

Also move it into the `case 'stream'` of the switch statement.